week-6/Homework/mandatory/1-alarmclock/index.html

```diff
@@ -24,5 +24,6 @@ <h1 id="timeRemaining">Time Remaining: 00:00</h1>
         <button id="stop" type="button">Stop Alarm</button>
       </div>
     </div>
+    <script src="alarmclock.js"></script>
```

Comment:
If you don't read elements immediately when the script is being executed first then it isn't necessary to be moved down. If elements are looked up inside functions that are invoked by button events then it won't throw any error because by that time page will have been fully loaded.

Reply:
I haven’t tried it before moving it to be honest, But one of the students posted about a problem she faced while solving the task and the solution was put the script to the bottom of the page. Thats why I went with it.

Reply:
That's okay. Just wanted to make sure you are aware of when a script fails and when not.

Basically, when the browser sees a script tag, it tries to execute that. But your functions are only called inside when something refers to and invokes them. It can be triggered by an event on a button, deferred call by a timer or of course you can call functions immediately.
But if the script tries to refer to something on the DOM that hasn't been loaded at that time then it will fail.

There are different technics to prevent errors. One can be what you did, moving the script tag to another place inside the HTML. Or you could see the `window.onload` event, it is also another way.

Reply:
@elsahib 
After reviewing others' code, I've noticed that you were right. It is a better approach, in this case, moving the script tag down because you must attach the event listener to the stop button. (Unfortunately, the `window.onload` event is already specified and it can't be added more than once and moreover the stipulation was to "do not modify the code below" 😄 .